src/utils/fees.ts

```diff
@@ -46,46 +60,132 @@ function isEIP1559Fees(chain: ChainId) {
   return chain === ChainId.Ethereum || chain === ChainId.Polygon || chain === ChainId.Avalanche;
 }
 
+/*
+ * Probable fee for EIP1559 (BASE + TIP)
+ */
+function probableFeePerUnitEIP1559(suggestedGasFee: EIP1559Fee) {
+  return suggestedGasFee.suggestedBaseFeePerGas || 0 + suggestedGasFee.maxPriorityFeePerGas;
+}
 
-async function getSendAmountFee(accountId: AccountId, asset: Asset, amount?: BN) {
-  const getMax = amount === undefined
-  const sendFees: {
-    [speed in keyof FeeDetails]: BN
-  } = { slow: new BN(0), average: new BN(0), fast: new BN(0), }
+/*
+ * Max fee for EIP1559
+ */
+function maxFeePerUnitEIP1559(suggestedGasFee: EIP1559Fee) {
+  return suggestedGasFee.maxFeePerGas;
+}
 
-  const assetFees = store.getters.assetFees(asset);
-  const feeAsset = getFeeAsset(asset) || getNativeAsset(asset)
-  if (assetFees) {
-    for (const [speed, fee] of Object.entries(assetFees)) {
-      const feePrice = fee.fee.maxPriorityFeePerGas + fee.fee.suggestedBaseFeePerGas || fee.fee
-      sendFees[speed as keyof FeeDetails] = getSendFee(feeAsset, feePrice)
-    }
-    if (asset === 'BTC') {
-      const client = store.getters.client({
-        network: store.state.activeNetwork,
-        walletId: store.state.activeWalletId,
-        asset: asset,
-        accountId: accountId
-      }) as Client<
-        BitcoinEsploraApiProvider,
-        BitcoinBaseWalletProvider
-      >;
-      const feePerBytes = Object.values(assetFees).map((fee) => fee.fee)
-      const value = getMax ? undefined : currencyToUnit(cryptoassets[asset], amount)
-      try {
-        const txs = feePerBytes.map((fee) => ({ value, fee }))
-
-        const totalFees = await client.wallet.getTotalFees(txs, getMax)
-        for (const [speed, fee] of Object.entries(assetFees)) {
-          const totalFee = unitToCurrency(cryptoassets[asset], totalFees[fee.fee])
-          sendFees[speed as keyof FeeDetails] = totalFee
-        }
-      } catch (e) {
-        console.error(e)
-      }
+/*
+ * Calculates fee for EIP1559 or returns fee for non EIP1559 chains per gas or byte (unit)
+ */
+function feePerUnit(suggestedGasFee: FeeType, chain: ChainId): number {
+  if (suggestedGasFee === undefined || chain === undefined) {
+    throw new Error('feePerUnit: Incorrect input!');
+  }
+
+  /*
+   * In case it is EIP1559 chain then `maxFeePerGas` will be set.
+   * Otherwise its non EIP1559 (this includes both EVM and Non EVM chains) and calculations are done based on `fee`.
+   */
+  if (isEIP1559Fees(chain) && (<EIP1559Fee>suggestedGasFee).maxFeePerGas) {
```

Comment:
There is way too much casting in this function

This is FeeType:
`export type FeeType = EIP1559Fee | number;`

Meaning you can first check if `suggestedGasFee` is a number and return,
Otherwise it is implicitly type `EIP1559Fee` and typescript will recognise it as such as long as other condition (number) has been handled before it

I believe this would remove all the casting in this function

Reply:
Ok, will remove casting.